pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -2218,13 +2252,27 @@ func (gce *GCECloud) DeleteDisk(diskToDelete string) error {
 // Builds the labels that should be automatically added to a PersistentVolume backed by a GCE PD
 // Specifically, this builds FailureDomain (zone) and Region labels.
 // The PersistentVolumeLabel admission controller calls this and adds the labels when a PV is created.
-func (gce *GCECloud) GetAutoLabelsForPD(name string) (map[string]string, error) {
-	disk, err := gce.getDiskByNameUnknownZone(name)
-	if err != nil {
-		return nil, err
+// If zone is specified, the volume will only be found in the specified zone,
+// otherwise all managed zones will be searched.
+func (gce *GCECloud) GetAutoLabelsForPD(name string, zone string) (map[string]string, error) {
+	var disk *gceDisk
+	var err error
+	if zone == "" {
+		disk, err = gce.getDiskByNameUnknownZone(name)
```

Comment:
Ah... so this is subtle.  This is actually a workaround for #27656 and #27657.  The problem is that in the volumes admission controller the cloudprovider is not configured with multizone=true.  This would prevent getDiskByNameUnknownZone from working at all unless the PD was in the master zone.  However, the workaround here is to allow creation in a non-master zone by looking to this zone parameter, which is derived by the admission controller when the zone annotation is already set.

Also, we actually want to be moving away getDiskByNameUnknownZone and towards volumes that always have an associated zone on GCE, because volume names on GCE are unique per zone, not per region.  i.e. the correct specifier for a GCE PD is probably `<zone>/<name>`, not `<name>`.  Thankfully with dynamic provisioning we can just use the annotation (which is painful for a human but fine for code), and static provisioning never really made sense on a cloud anyway...

I guess this needs a comment!